SourceryTests/Parsing/ComposerSpec.swift

```diff
@@ -396,6 +403,32 @@ class ParserComposerSpec: QuickSpec {
                     }
                 }
 
+                context("given generic inheritance") {
+                    it("resolves inherited type for subclasses") {
+                        let result = parse("""
+                            class Generic<T> {}
+                            class Specialized : Generic<Int> {}
+                        """)
+                        let generic = Class(name: "Generic", genericTypePlaceholders: [ GenericTypePlaceholder(placeholderName: TypeName("T")) ])
+                        let specialized = Class(name: "Specialized", inheritedTypes: ["Generic<Int>"])
+                        expect(result).to(equal([ generic, specialized]))
+                    }
+
+                    it("resolves inherited types for generic placeholders") {
+                        let result = parse("""
+                            class Generic<T> {}
```

Comment:
can we add a test where there is a constraint here as well, i.e. `T: Bar` where `protocol Bar {}`